File: back_end/app/routes/saved_listings.py
from fastapi import APIRouter
from app.services.saved_listing_service import toggle_saved_listing, get_saved_listings
from app.core.firebase import db

router = APIRouter(prefix="/saved-listings", tags=["saved-listings"])


@router.post("/toggle")
def toggle(userId: str, listingId: str):
    return toggle_saved_listing(db, userId, listingId)


@router.get("")
def get_saved(userId: str):
    return get_saved_listings(db, userId)

File: front_end/src/api/savedListings.ts
const API_BASE = import.meta.env.VITE_API_BASE_URL ?? "http://localhost:8000";

export async function toggleSavedListing(userId: string, listingId: string) {
  const response = await fetch(
    `${API_BASE}/saved-listings/toggle?userId=${encodeURIComponent(userId)}&listingId=${encodeURIComponent(listingId)}`,
    {
      method: "POST",
    }
  );

  if (!response.ok) {
    throw new Error("Failed to toggle saved listing");
  }

  return response.json();
}

export async function getSavedListings(userId: string) {
  const response = await fetch(
    `${API_BASE}/saved-listings?userId=${encodeURIComponent(userId)}`
  );

  if (!response.ok) {
    throw new Error("Failed to fetch saved listings");
  }

  return response.json();
}
